apps/explorer/src/app/components/leverage/leverage.tsx

```diff
@@ -0,0 +1,39 @@
+import { Tooltip } from '@vegaprotocol/ui-toolkit';
+import BigNumber from 'bignumber.js';
+import { LeverageTooltip } from './leverage-tooltip';
+
+export type LeverageProps = {
+  marginFactor?: string;
+  className?: string;
+};
+
+/**
+ * Calculates leverage, given a margin factor, protecting for 0 or null
+ * or infinity
+ * @returns React component
+ */
+export function Leverage({ marginFactor, className = '' }: LeverageProps) {
+  if (!marginFactor || parseFloat(marginFactor) === 0) {
+    return null;
+  }
+
+  BigNumber.config({ DECIMAL_PLACES: 2 });
+
+  const leverage = BigNumber(1).dividedBy(BigNumber(marginFactor)).toString();
+
+  if (leverage === 'Infinity' || leverage.indexOf('e+') !== -1) {
```

Comment:
BigNumber has `isFinite` method that could be used here. There's no need to check the `string` value of that number.

Also what is the reason of checking for the exponential here?

Reply:
Ok, I assume if it's close to infinity then ignore, but you could just cap this to some reasonable number rather than relying on the exponential formatting.

Reply:
Btw that's the default formatting cap `999999999999999900000`

Reply:
I checked with @davidsiska-vega and he suggested a much lower cap, so I doubled *his* cap and got... 1001. I'm going with that.